### PR TITLE
[Dart] Avoid redundant null checking in fromJson

### LINKED ISSUE #3628: [REQ] [Dart] Remove redundant null checks
### Is your feature request related to a problem? Please describe.

The fromJson methods in my models have lots of redundant checks like this: 
```Dart
if (json['stop_distance'] == null) {
  stopDistance = null;
} else {
  stopDistance = json['stop_distance'];
}
```

## Describe the solution you'd like

Remove redundant check and just assign:
```Dart
stopDistance = json['stop_distance'];
```

## Describe alternatives you've considered

I'll start a PR now, should just be a small change to `class.mustache` I believe.

### PR DESCRIPTION
**Pull request by nickmeinhold**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
  - @ircecho @swipesight @jaumard

### Description of the PR

fix #3628 

- changed class.mustache so that the null check is only put in when there is a reason to 
- the mustache file is now more verbose as the null check is repeated for each case but it seemed the simplest way to make the change without trying to mess around with the logic of the template

### CODE REVIEW

File: samples/client/petstore/dart/flutter_petstore/openapi/lib/model/api_response.dart [hunk 1]
```diff
@@ -2,11 +2,11 @@ part of openapi.api;
 
 class ApiResponse {
   
-  int code = null;
+    int code = null;
```
Comment (wing328) [inline samples/client/petstore/dart/flutter_petstore/openapi/lib/model/api_response.dart]: @nickmeinhold looks like there are 2 additional space inserted. Is that intentional?

Comment (nickmeinhold) [reply samples/client/petstore/dart/flutter_petstore/openapi/lib/model/api_response.dart]: No, not intentional - I didn't notice that sorry.  At first glance I don't understand where the extra spaces have come from as the only change I made to the template was inside the fromJson part. Any thoughts?  I will have a look tonight and see if I can understand it.

Comment (nickmeinhold) [reply samples/client/petstore/dart/flutter_petstore/openapi/lib/model/api_response.dart]: Yeah I really don't get this - these lines are presumably generated by line4 of `class.mustache` and I haven't changed it or anything else nearby.  I'm still learning mustache, there must be something about how the indentation works that I'm missing.  I will read up more on mustache when I get some time, probably won't be today though.

Comment (wing328) [reply samples/client/petstore/dart/flutter_petstore/openapi/lib/model/api_response.dart]: Don't worry. I'll take care of the formating with another PR

**Comment by wing328** [comment]
@nickmeinhold thanks for the PR, which has been included in the v4.1.1 release: https://twitter.com/oas_generator/status/1165944867391860737

**Pull request merged**